loopy/match.py

```diff
@@ -243,8 +243,19 @@ def __call__(self, kernel, matchable):
 
 class Tagged(GlobMatchExpressionBase):
     def __call__(self, kernel, matchable):
+        from loopy.kernel.instruction import LegacyStringInstructionTag
         if matchable.tags:
-            return any(self.re.match(tag) for tag in matchable.tags)
+            return any(
+                    self.re.match(tag.value)
+                    if isinstance(tag, LegacyStringInstructionTag)
+                    else
+
+                    self.re.match(tag) if isinstance(tag, str)
```

Comment:
InstructionBase's docs suggest that tags are instances `pytools.tag.Tag`, in that case this else branch shouldn't exist.

Reply:
These matchers also work on `SubstitutionRule`, which hasn't made the jump yet. You're right, maybe it should.

Reply:
I lied. It's really `TaggedVariable` that needs to be changed.

Reply:
035e318f